tigrisdb/vector_store.py

```diff
@@ -0,0 +1,77 @@
+from typing import List, Optional
+
+import grpc
+
+from tigrisdb.client import TigrisClient
+from tigrisdb.errors import TigrisServerError
+from tigrisdb.types import ClientConfig
+from tigrisdb.types.filters import Filter
+from tigrisdb.types.search import DocStatus, IndexedDoc, Query, Result, VectorField
+from tigrisdb.types.vector import Document, DocWithScore
+
+
+class VectorStore:
+    def __init__(self, name: str, config: Optional[ClientConfig] = None):
+        self.client = TigrisClient(config).get_search()
+        self._index_name = name
+        self.index = self.client.get_index(name)
+
+    @property
+    def name(self):
+        return self._index_name
+
+    def _ensure_index(self, dimension: int):
+        self.client.create_or_update_index(
+            name=self.name,
+            schema={
+                "title": self.name,
+                "additionalProperties": False,
+                "type": "object",
+                "properties": {
+                    "id": {"type": "string"},
+                    "text": {"type": "string"},
+                    "metadata": {"type": "object"},
+                    "embeddings": {
+                        "type": "array",
+                        "format": "vector",
+                        "dimensions": dimension,
+                    },
+                },
+            },
+        )
+
+    def add_documents(self, docs: List[Document]) -> List[DocStatus]:
+        # try inserting, if index is not found, ensure index and insert again
+        try:
+            return self.index.create_many(docs)
```

Comment:
Is `id` getting generated on the server side? Or can user pass it? If the user can pass it, let's also add a comment about the behavior of `add_documents`